spec/elastic_apm/transport/serializers/span_serializer_spec.rb

```diff
@@ -66,6 +68,29 @@ module Serializers
               expect(result.dig(:span, :context, :db, :statement))
                 .to eq 'asd'
               expect(result.dig(:span, :context, :http, :url)).to eq 'dsa'
+              expect(result.dig(:span, :context, :sync)).to eq false
+            end
+
+            context 'when sync is nil' do
+              let(:span) do
+                Span.new(
+                  name: 'Span',
+                  transaction: transaction,
+                  parent: transaction,
+                  trace_context: trace_context,
+                  context: Span::Context.new(
+                    db: { statement: 'asd' },
+                    http: { url: 'dsa' }
+                  )
+                )
+              end
+
+              it 'adds context object' do
+                expect(result.dig(:span, :context, :db, :statement))
+                  .to eq 'asd'
+                expect(result.dig(:span, :context, :http, :url)).to eq 'dsa'
+                expect(result.dig(:span, :context, :sync)).to be nil
```

Comment:
It might be more interesting to check that `span.context` doesn't have a `:sync` key?
It's essentially the same but tells more about the implementation.